// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package SafeEntry;

// The greeting service definition.
service SafeEntry {
  rpc CheckIn (Request) returns (Reply) {}
  rpc CheckOut (Request) returns (Reply) {}
  rpc CheckInHistory (Request) returns (Reply) {}
}

// The request message containing the user's name.
message Request1 {
  int32 x = 1;
  int32 y = 2;
}

message Request {
  string name = 1;
  string nric = 2;
  string location = 3;
  string datetime = 4;
  string status = 5;
}

// The response message containing the greetings
message List {
  string name = 1;
  string nric = 2;
  string location = 3;
  string datetime = 4;
  string status = 5;
}

message ReplyList {
  repeated List res = 1;
}

message Reply {
  string res = 1;
}
